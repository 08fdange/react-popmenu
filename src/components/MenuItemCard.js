import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Button from './buttons/Button';
import withAppManager from '../utilities/withAppManager';
import Colors from '../theme/colors';

const propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  price: PropTypes.number,
  imgUrl: PropTypes.string,
  viewport: PropTypes.string,
  handleDelete: PropTypes.func,
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: ${({ isMobile }) => isMobile ? 'column' : 'row'};
  padding: 20px 16px;
  border-top: 1px solid #e6e6e6;
  border-bottom: 1px solid #e6e6e6;
`;

const ImgWrapper = styled.div`
  display: flex;
`;

const StyledImg = styled.img`
  height: ${({ isMobile }) => isMobile ? '100%' : '120px'};
  width: ${({ isMobile }) => isMobile ? '100%' : '120px'};
  aspect-ratio: 1 / 1;
  border-radius: 6px;
  vertical-align: middle;
  margin-right: 16px;
  ${({ isMobile }) => isMobile && 'margin-bottom: 16px;'}
`;

const TextSection = styled.div`
  display: flex;
  flex-direction: column;
`;

const TitleText = styled.h2`
  margin: 0;
  margin-bottom: 12px;
  padding: 0;
`;

const DescriptionText = styled.p`
  margin: 0;
  padding: 0;
`;

const PriceSection = styled.div`
  display: flex;
  flex-direction: ${({ isMobile }) => isMobile ? 'row' : 'column'};
  justify-items: space-between;
  margin-left: ${({ isMobile }) => isMobile ? '0' : 'auto'};
  padding-left: ${({ isMobile }) => isMobile ? '0' : '16px'};
  ${({ isMobile }) => isMobile && 'padding-top: 16px;'}
`;

const PriceText = styled.h3`
  margin: 0;
  padding: 0;
  flex: 1;
`;

const MenuItemCard = (props) => {
  const {
    title,
    description,
    price,
    imgUrl,
    viewport,
    handleDelete,
  } = props;

  const isMobile = viewport === 'mobile' || viewport === 'mobileSmall';

  return (
    <Wrapper isMobile={isMobile} {...props}>
      <ImgWrapper>
        <StyledImg isMobile={isMobile} src={imgUrl} alt={title} />
      </ImgWrapper>
      <TextSection>
        <TitleText>{title}</TitleText>
        <DescriptionText>{description}</DescriptionText>
      </TextSection>
      <PriceSection isMobile={isMobile}>
        <PriceText>{`$${price.toFixed(2)}`}</PriceText>
        <Button
          onClick={handleDelete}
          backgroundColor={Colors.SecondaryRed}
          hoverBackgroundColor={Colors.SecondaryRedDark}
        >Delete</Button>
      </PriceSection>
    </Wrapper>
  );
};

MenuItemCard.propTypes = propTypes;

export default withAppManager(MenuItemCard);

