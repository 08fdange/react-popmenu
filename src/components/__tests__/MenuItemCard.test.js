import React from 'react';
import renderer from 'react-test-renderer';
import MenuItemCard from '../MenuItemCard';

const testData = {
  id: 0,
  title: 'Bologna Sandwich',
  description: 'Fried bologna sandwich with cheese that\'s likely past it\'s expiration date.',
  price: 1,
  imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/2/20/Bolgona_sandwich.jpg'
}

test('MenuItemCard - renders correctly', () => {
  const tree = renderer.create(<MenuItemCard {...testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('MenuItemCard - renders props correctly', () => {
  const testRenderer = renderer.create(<MenuItemCard {...testData} />);
  const testInstance = testRenderer.root;
  expect(testInstance.props.title).toBe('Bologna Sandwich');
  expect(testInstance.props.imgUrl).toBe('https://upload.wikimedia.org/wikipedia/commons/2/20/Bolgona_sandwich.jpg');
  expect(testInstance.props.price).toBe(1);
});