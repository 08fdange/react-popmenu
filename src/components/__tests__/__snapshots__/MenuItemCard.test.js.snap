// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemCard MenuItemCard - renders correctly 1`] = `
<div
  className="sc-hKwDye kQLoSl"
  id={0}
  title="Bologna Sandwich"
>
  <div
    className="sc-eCImPb bqIGb"
  >
    <img
      alt="Bologna Sandwich"
      className="sc-jRQBWg ePIiRZ"
      src="https://upload.wikimedia.org/wikipedia/commons/2/20/Bolgona_sandwich.jpg"
    />
  </div>
  <div
    className="sc-gKclnd gUQszg"
  >
    <h2
      className="sc-iCfMLu eYrbJW"
    >
      Bologna Sandwich
    </h2>
    <p
      className="sc-furwcr bgEzXB"
    >
      Fried bologna sandwich with cheese that's likely past it's expiration date.
    </p>
  </div>
  <div
    className="sc-pVTFL bKhZfx"
  >
    <h3
      className="sc-jrQzAO civLCO"
    >
      $1.00
    </h3>
    <button
      className="sc-bdvvtL htfPGY"
    >
      Delete
    </button>
  </div>
</div>
`;
